lib/rules/no-setup-props-destructure.js

```diff
@@ -75,6 +91,34 @@ module.exports = {
         report(left, 'getProperty', propsReferences.scopeName)
       }
     }
+    /**
+     * @param {Expression} node
+     * @param {ScopePropsReferences} propsReferences
+     */
+    function isPropsMemberAccessed(node, propsReferences) {
+      const propRefs = [...propsReferences.refs.values()]
+      const isMemberAccessed = propRefs.some((props) => {
+        const isPropsInExpressionRange = utils.inRange(node.range, props)
+
+        if (isPropsInExpressionRange) {
+          const isPropsMemberExpression =
+            props.parent.type === 'MemberExpression' &&
+            props.parent.object === props
+
+          if (isPropsMemberExpression) {
+            return true
+          }
+        }
+
+        return false
+      })
+
+      if (isMemberAccessed) {
+        return true
+      }
+
+      return false
```

Comment:
```suggestion
      return propRefs.some((props) => {
        const isPropsInExpressionRange = utils.inRange(node.range, props)

        if (isPropsInExpressionRange) {
          const isPropsMemberExpression =
            props.parent.type === 'MemberExpression' &&
            props.parent.object === props

          if (isPropsMemberExpression) {
            return true
          }
        }

        return false
      })
```

Reply:
Thanks for the suggestion. 💯 I cleaned it up a bit to a point where it's still readable but much more concise.